hw/ip/spi_device/doc/_index.md

```diff
@@ -412,6 +472,53 @@ instance, if FW sets RXFIFO depth to 128 (default value), it should not update
 the read pointer outside the range 0x000 -  0x1FF (128*4 = 512Bytes ignoring
 the phase bit, bit 11).
 
+## TPM over SPI
+
+### Initialization
+
+The SW should enable the TPM submodule by writing 1 to the TPM_CFG.en CSR field.
+Other SPI_DEVICE features (Generic, Flash, Passthrough) CSRs do not affect the TPM feature.
+
+Update TPM_ACCESS_0, TPM_ACCESS_1 CSRs.
+The TPM submodule uses TPM_ACCESS_x.activeLocality to determine if the TPM_STS is returned to the SB.
+The SW may configure TPM_CFG.hw_reg_dis and/or TPM_CFG.invalid_locality to fully control the TPM transactions.
+
+### TPM mode: FIFO and CRB
+
+The HW returns the return-by-HW registers for the read request in the TPM FIFO mode.
+In the TPM CRB mode (TPM_CFG.tpm_mode is 1), the logic always upload the command and address to the SW and waits for the read FIFO data even the received address falls into the managed address.
+
+### Return-by-HW register update
+
+The SW manages the retun-by-HW registers.
+The contents are placed inside the SPI_DEVICE CSRs.
+The SW must maintain the other TPM registers outside of the SPI_DEVICE HWIP and uses write/read FIFOs to receive the content from/ send the register value to the SB.
+
+When the SW updates the return-by-HW registers, the SW is recommended to read back the register to confirm the value is written.
+Due to the CDC issue, the SW is permitted to update the registers when the TPM CS# is de-asserted.
+
```

Comment:
do you mean software is only permitted to update when cs is de-asserted? or that software can update also when cs is de-asserted?

Reply:
Added "only"